### ISSUE TITLE
Support for parametric geometry descriptions via links between a non-geometric property and geometric properties

### ISSUE THREAD

**Issue by FransKnibbe** [opengeospatial/ogc-geosparql#8]
Contributed by Anna Wagner and Mathias Bonduel for the white paper "Benefits of Representing Spatial Data using Semantic and Graph Technologies":

During the design phase of a building, a series of geometric descriptions are made based on input parameters (lengths, size, location, orientation). These parameters are commonly also part of the non-geometric description, causing redundant information. In traditional descriptions, such as the Industry Foundation Classes (IFC), this can lead to the non-geometric description contradicting the geometric one, since they are not updated accordingly. For example, a wall can be defined in IFC, where the height of the wall is a non-geometric property, while the extrusion of the geometry is stored separately. The non-geometric property is neither connected to the extrusion nor automatically updated, when the geometry changes, resulting in the potential of inconsistent data. Thus, a link between related properties from geometric and non-geometric descriptions should be established, to ease the detection of contradicting data and the subsequent updating process.

In the Ontology for Managing Geometry (OMG, https://w3id.org/omg), a first implementation of such links is realised. For one, the omg:isExplicitlyDerivedFrom property can be applied in cases, a geometric property of RDF-based geometry (object) and a non-geometric (subject) property describe exactly the same situation (e.g. the height/extrusion), where a chain axiom can be used to automatically update the according values. On the other hand, the omg:isImplicitlyDerivedFrom property can be applied to indicate that a non-geometric property (subject) can be derived from a geometry description (object), as is the case for volumes or surface areas.

- Remark: adding metadata regarding the used coordinate system, units and other metadata as mentioned in extension 1 and 4 can also result in double information.  Some geometry schemas such as OBJ are unitless so additional info on the length unit is useful, while others allow to define the used units internally (e.g. STEP) or have a fixed length unit defined in the schema (e.g. glTF uses metre). Adding metadata in RDF for OBJ geometry description enriches the knowledge over the geometry. If this is also added for STEP or glTF geometry, this results in redundant data, but the metadata is now externalized in RDF can thus be queried directly (no interpretation needed).

**Comment by jabhay** [comment]
MoSCoW poll created

[![](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Must)](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Must/vote)
[![](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Should)](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Should/vote)
[![](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Could)](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Could/vote)
[![](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Won%27t)](https://api.gh-polls.com/poll/01EJXBFY7CN0JS4J0AJAC3KHJY/Won%27t/vote)

**Comment by mathib** [comment]
For GeoSPARQL 1.1 revision, it would be useful to define links between spatial properties (lenght, area, volume as currently introduced) and the geometry description it might be derived from (e.g. geometry resulting directly from a spatial survey: a point cloud, mesh, set of coordinates, etc.). To achieve this, `prov:wasDerivedFrom` can be reused (which is rather generic) or it can be further specialised, e.g. into `geo:isDerivedPropertyFrom` with domain equal to `geo:SpatialMeasure` and range equal to `geo:Geometry`.

Note that these spatial properties are to be attached to the geo:Feature instance and not the geo:Geometry nodes (see domain of `geo:hasLength` etc.).

**Comment by mathib** [comment]
this issue also contains a part regarding RDF-based geometry (scheduled for GeoSPARQL 2.0), where properties are an inherent part of the geometry description itself and therefore this issue is best split in two parts I guess.

**Comment by FransKnibbe** [comment]
@mathib: It seems that inclusion of `geo:SpatialMeasure` in GeoSPARQL 1.1 for a large part satisfies this change request. Would agreement on how to model provenance make it complete?

I think having a way to express provenance of a `geo:SpatialMeasure` would be welcome. But provenance could be more than another geometry, I think. For example, I could see a tree or a building somewhere and just estimate its height.

**Comment by mathib** [comment]
@FransKnibbe : no, it's only addressed to a very limited extend now, in GeoSPARQL 1.1. If there would be a movement to include the definition RDF-based geometry descriptions (such as GEOM, OntoBREP, OntoSTEP, OCC,etc.), there could be options to do deep linking of parts of the geometry that correspond (e.g. a height parameter that's intrisincly part of the geometry) with a `geo:SpatialMeasure` related to a feature of interest. 

Provenance for `geo:SpatialMeasure` can indeed be broader than just linking back to the geometry where the measure was taken from. Examples are accuracy, links between derived properties (e.g. area derived from a width and length), link to a survey campaign, etc.
